File: app/client/Contrib.Gate/Assets/Application/Entities/Vending.cs
using CsvHelper.Configuration;

namespace Entities
{
    public class Vending
    {
        public Identify Identify;
        public Identify Result;
        public int Level;
        public int Price;
        public int Num;
    }

    public sealed class VendingMap : ClassMap<Vending>
    {
        public VendingMap()
        {
            Map(x => x.Identify).Index(0);
            Map(x => x.Result).Index(1);
            Map(x => x.Level).Index(2);
            Map(x => x.Price).Index(3);
            Map(x => x.Num).Index(4);
        }
    }
}

File: app/server/Contrib.Gate/application/entities/Vending.ts
// 自販機データ
namespace Entities {
    export class Vending {
        private bucket: Bucket<Vending>;
        private identify: Identify;

        constructor(admin: KiiAppAdminContext, identify: Identify) {
            this.bucket = new Bucket(this, admin, "vending");
            this.identify = identify;
        }

        refresh(done: (vending: Vending) => void) {
            let clause = KiiClause.equals("ID", this.identify.toString());
            this.bucket.refresh(done, KiiQuery.queryWithClause(clause));
        }

        get target(): Identify {
            return Identify.Parse(this.bucket.first.get("購入対象ID"));
        }
        get level(): number {
            return this.bucket.first.get("購入レベル");
        }
        get price(): number {
            return this.bucket.first.get("値段");
        }
        get num(): number {
            return this.bucket.first.get("個数");
        }
    }
}